astropy/coordinates/angles/core.py

```diff
@@ -415,14 +415,17 @@ def _wrap_at(self, wrap_angle):
         # Do the wrapping, but only if any angles need to be wrapped
         #
         # Catch any invalid warnings from the floor division.
-        with np.errstate(invalid="ignore"):
-            wraps = (self_angle - wrap_angle_floor) // a360
-        valid = np.isfinite(wraps) & (wraps != 0)
-        if np.any(valid):
-            self_angle -= wraps * a360
-            # Rounding errors can cause problems.
-            self_angle[self_angle >= wrap_angle] -= a360
-            self_angle[self_angle < wrap_angle_floor] += a360
+
+        out_of_range = (self_angle < wrap_angle_floor) | (self_angle >= wrap_angle)
```

Comment:
Actually, given that we were confused about the use of `np.isfinite` and read-only arrays, perhaps we should have a comment here along the lines of,
```
# See if any wrapping is necessary. It is useful to avoid
# this since the array can be read-only due to broadcasting.
# Note that since comparisons with NaN are always False,
# this also ensures that no adjustments are made for a 
# read-only array with some NaN but otherwise OK elements.
```
(Feel free to adjust!  And perhaps wait for all the tests to pass,
so this can be a `skip ci` comment update.)

Reply:
Actually let's run CI again: the results I got from the benchmark are *so* good that I'd like to double check !